querydsl-jpa/src/main/java/com/querydsl/jpa/HibernateHandler.java

```diff
@@ -52,8 +52,8 @@ public boolean createNativeQueryTyped() {
     @SuppressWarnings("unchecked")
     @Override
     public <T> CloseableIterator<T> iterate(Query query, FactoryExpression<?> projection) {
-        if (query instanceof NativeQuery) {
-            NativeQuery hQuery = (NativeQuery) query;
+        if (query instanceof org.hibernate.query.Query) {
```

Comment:
Shouldnt we just always attempt to unwrap and not do any instanceof checks at all?